x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -252,6 +252,10 @@ func (r *metricsRequester) getFilterForMetric(serviceName, m string) string {
 	// to apply any additional filters.
 	if locationsConfigsAvailable && !isAGlobalService(serviceName) {
 		serviceLabel := getServiceLabelFor(serviceName)
+		// If user has entered a location label, overide the serviceLabel with the locationlabel
+		if r.config.LocationLabel != "" {
```

Comment:
Rather than checking if empty, maybe to check exact mathes of string?

Reply:
There is not an exhaustive list of these labels. We want to give the flexibility to the user to overwrite it as this is a generic input. 
We have metricsets for specific services with the specific labels. Eg: compute, redis etc. 
So, IMO we can just take in the input from user as it is. It gives us the flexibility to not change beats code in case GCP services make some changes in their labels in future. We can handle that from the Integration directly.